dotnet: migrate nuget lockfiles to JSON (#362278)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2505.section.md

- `networking.wireguard` now has an optional networkd backend. It is enabled by default when `networking.useNetworkd` is enabled, and it can be enabled alongside scripted networking with `networking.wireguard.useNetworkd`. Some `networking.wireguard` options have slightly different behavior with the networkd and script-based backends, documented in each option.

> - `buildDotnetModule` now uses JSON-based lockfiles instead of nix. Nix-based lockfiles were kept for compatibility however they should be migrated to JSON.

Reviewer: Some ideas for alternative wording:

```suggestion
- `buildDotnetModule` now uses JSON-based instead of nix-based lockfiles.
  - Support for nix-based lockfiles was kept for compatibility, however this support will be dropped in the future.
```

- `networking.wireguard` now has an optional networkd backend. It is enabled by default when `networking.useNetworkd` is enabled, and it can be enabled alongside scripted networking with `networking.wireguard.useNetworkd`. Some `networking.wireguard` options have slightly different behavior with the networkd and script-based backends, documented in each option.

- `buildDotnetModule` now uses JSON-based instead of nix-based lockfiles.

>   Support for nix-based lockfiles was kept for compatibility, however this support will be dropped in the future.

Author: Deprecated can be a shorter word for it, however I prefer making it explicit we don't have plans to keep it around, considering the amount of things in the tech world that are _technically_ deprecated but stay around for ages.

>   Support for nix-based lockfiles was kept for compatibility, however this support will be dropped in the future.

Reviewer 2: well, you can say 'deprecated, and will be removed in release XX.YY'.

it's a common term that's used throughout the docs

>   Support for nix-based lockfiles was kept for compatibility, however this support will be dropped in the future.

Author: Okay, will do that!
